### ISSUE TITLE
tpm2_ptool depends on tpm2-tools utlities not checked in configure

### ISSUE THREAD

**Issue by bnevis-i** [tpm2-software/tpm2-pkcs11#562]
Dependency on tpm2-tools should have been caught during configure.  Dependency on pyasn1_modules should have also been caught.  Ideally no dependency on tpm2-tools at all -- use TSS library C functions.

## Environment

Board: APL UPSquared with PTT
OS: Ubuntu 20.04 (opensc 0.20.0-3)
tpm2-tss: 4.2.1 (tarball)
tpm2-pkcs11: 1.3.1 (tarball)
tpm2-tools: Not installed
tpm2-abrmd: Not installed

## Error message
```
./tpm2_ptool init 
Traceback (most recent call last):
  File "/home/bnevis/tpm2-pkcs11-1.3.1/tools/tpm2_pkcs11/commandlets_store.py", line 86, in __call__
    ctx = tpm2.createprimary(ownerauth, pobjauth)
  File "/home/bnevis/tpm2-pkcs11-1.3.1/tools/tpm2_pkcs11/tpm2.py", line 29, in createprimary
    p = Popen(cmd, stdout=PIPE, stderr=PIPE, env=os.environ)
  File "/usr/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'tpm2_createprimary'
[Errno 2] No such file or directory: 'tpm2_createprimary'
```

**Comment by williamcroberts** [comment]
> Dependency on tpm2-tools should have been caught during configure. Dependency on pyasn1_modules should have also been caught. Ideally no dependency on tpm2-tools at all -- use TSS library C functions.

This is captured by bug #129 (use python bindings). It's not super trivial, as the tools do a ton of work before actually calling the tss routines.

> 
> ## Environment
> Board: APL UPSquared with PTT
> OS: Ubuntu 20.04 (opensc 0.20.0-3)
> tpm2-tss: 4.2.1 (tarball)
> tpm2-pkcs11: 1.3.1 (tarball)
> tpm2-tools: Not installed
> tpm2-abrmd: Not installed
> 
> ## Error message
> ```
> ./tpm2_ptool init 
> Traceback (most recent call last):
>   File "/home/bnevis/tpm2-pkcs11-1.3.1/tools/tpm2_pkcs11/commandlets_store.py", line 86, in __call__
>     ctx = tpm2.createprimary(ownerauth, pobjauth)
>   File "/home/bnevis/tpm2-pkcs11-1.3.1/tools/tpm2_pkcs11/tpm2.py", line 29, in createprimary
>     p = Popen(cmd, stdout=PIPE, stderr=PIPE, env=os.environ)
>   File "/usr/lib/python3.8/subprocess.py", line 854, in __init__
>     self._execute_child(args, executable, preexec_fn, close_fds,
>   File "/usr/lib/python3.8/subprocess.py", line 1702, in _execute_child
>     raise child_exception_type(errno_num, err_msg, err_filename)
> FileNotFoundError: [Errno 2] No such file or directory: 'tpm2_createprimary'
> [Errno 2] No such file or directory: 'tpm2_createprimary'
> ```

I think we got them all on the tip of master, which will be the next release. Do you need a backport to 1.3.X?

**Comment by bnevis-i** [comment]
> Do you need a backport to 1.3.X?

No. This is a kicking-the-tires exercise and I wanted to make sure I used a known labeled version.  I will try tip and see how it does.

**Comment by bnevis-i** [comment]
PS It would help to have a README.md at the top level of the project rather than burying everything in docs/

**Comment by williamcroberts** [comment]
> PS It would help to have a README.md at the top level of the project rather than burying everything in docs/

Github pulls it out fine, I can add a symlink...

See https://github.com/tpm2-software/tpm2-pkcs11/pull/574

**Comment by williamcroberts** [comment]
Ill back-port the configure checks to 1.3.2 since we have a release going for the segfault.

**Comment by williamcroberts** [comment]
Fixed in #575
